lib/rides/pages/search_suggestion_page.dart

```diff
@@ -309,19 +349,274 @@ class _SearchSuggestionPage extends State<SearchSuggestionPage> {
+      ),
+    );
+  }
+
+  Widget _buildSortingFilter(void Function(void Function()) innerSetState) {
+    return filterCategory(
+      S.of(context).pageSearchSuggestionsFilterSorting,
+      DropdownButton(
+        value: _sorting,
+        items: SearchSuggestionSorting.values
+            .map(
+              (sorting) => DropdownMenuItem(
+                value: sorting,
+                child: Text(sorting.getDescription(context)),
+              ),
+            )
+            .toList(),
+        onChanged: (SearchSuggestionSorting? value) => innerSetState(
+          () => _sorting = value!,
+        ),
+      ),
+    );
+  }
+
+  void _showFilterDialog() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) => StatefulBuilder(
+        builder: (context, innerSetState) {
+          return AlertDialog(
+            content: SingleChildScrollView(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  _buildRatingFilter(innerSetState),
+                  _buildFeaturesFilter(innerSetState),
+                  _buildDeviationFilter(),
+                  _buildSortingFilter(innerSetState),
+                ],
+              ),
+            ),
+            actions: <Widget>[
+              TextButton(
+                child: Text(S.of(context).pageSearchSuggestionsFilterResetToDefault),
+                onPressed: () => innerSetState(() => setDefaultFilterValues()),
+              ),
+              TextButton(
+                child: Text(S.of(context).okay),
+                onPressed: () {
+                  setState(() {});
+                  Navigator.of(context).pop();
+                },
+              ),
+            ],
+          );
+        },
+      ),
+    );
+  }
+
+  List<Ride> _applyFilters(List<Ride> rideSuggestions) {
+    return rideSuggestions
+        .where(
+          (Ride ride) {
+            Profile driver = ride.drive!.driver!;
+            bool ratingSatisfied = driver.getAggregateReview().rating >= _minRating &&
+                driver.getAggregateReview().comfortRating >= _minComfortRating &&
+                driver.getAggregateReview().safetyRating >= _minSafetyRating &&
+                driver.getAggregateReview().reliabilityRating >= _minReliabilityRating &&
+                driver.getAggregateReview().hospitalityRating >= _minHospitalityRating;
+            bool featuresSatisfied = Set.of(driver.profileFeatures!).containsAll(_selectedFeatures);
+            bool maxDeviationSatisfied =
+                _selectedDate.difference(ride.startTime) < Duration(hours: int.parse(_maxDeviationController.text));
+            return ratingSatisfied && featuresSatisfied && maxDeviationSatisfied;
+          },
+        )
+        .sorted(_sorting.sortFunction(_selectedDate))
+        .toList();
+  }
+}
+
+enum SearchSuggestionSorting { relevance, time, price }
```

Comment:
I think price is good. I would change time to traveltime. And relevance is the price and traveltime together?